api/repositories/service_instance_repository_test.go

```diff
@@ -428,10 +422,134 @@ var _ = Describe("ServiceInstanceRepository", func() {
 			})
 		})
 	})
+
+	Describe("GetServiceInstance", func() {
+		var (
+			space2          *v1alpha2.SubnamespaceAnchor
+			serviceInstance *servicesv1alpha1.CFServiceInstance
+			record          repositories.ServiceInstanceRecord
+			getErr          error
+			getGUID         string
+		)
+
+		BeforeEach(func() {
+			space2 = createSpaceAnchorAndNamespace(testCtx, org.Name, prefixedGUID("space2"))
+
+			serviceInstance = createServiceInstance(space.Name, "the-service-instance")
+			createServiceInstance(space2.Name, "some-other-service-instance")
+
+			getGUID = serviceInstance.Name
+		})
+
+		JustBeforeEach(func() {
+			record, getErr = serviceInstanceRepo.GetServiceInstance(testCtx, authInfo, getGUID)
+		})
+
+		It("returns a forbidden error", func() {
```

Comment:
Nit: Even though there is no additional set up needed, I think it would improve clarity if we wrapped this in a `When("the user does not have permissions to get the service instance", ...`

Reply:
I would strongly vote against such contexts as they do not contribute to making the test environment more specific, instead they are (ab)used for documentation purposes. If documenting user not being allowed to do that is really desired, I would prefer putting that into the `It` description, e.g. `It("returns a forbidden error (as the user is not allowed to do that)....`

Reply:
🤷 

Maybe it's due to my background in the Ruby community, but the most common style guides recommend this. I read  code better when I can scan up and down through the contexts to see what is being tested, but I see how this can be a personal preference.

What I'm suggesting is what is recommended by these guidelines and I generally agree:
* https://www.betterspecs.org/#contexts
* https://www.betterspecs.org/#short

That said, I also don't like to see deeply nested contexts for no reason, so it's definitely a judgement call. I don't think what I suggested _needs_ to be done in this case, but I also wouldn't want to see a hard rule against what I'm suggesting because I personally find it better. 🙂

Reply:
The closes example that I can think of is [this](https://medium.com/swlh/effective-ginkgo-gomega-b6c28d476a09#cb3d) by one of Ginkgo's maintainers. I would suggest that you put a comment in the ginkgo style guide proposal, since it sounds related

Reply:
Yeah I added a comment on the proposal yesterday 👍 

My interpretation of that blog post is that it's centered around the role of `BeforeEach` and less around using `Describe` and `Context` to provide test clarity.

Overall my thinking here though is that it's not a huge deal either way and I don't think there is much value in policing this issue across the whole codebase. I think either way has the potential to result in readable and functioning code so it's ok for something like this to differ based on who wrote the test.

I think on the proposal I agreed with the particular example that something like `When("on the happy path", ...` is not helpful and should be avoided, but that if someone adds a test and has something like `When("the user is unauthorized", ...` and does not need to provide additional setup because it's a lack of setup then that should be perfectly fine to do so. On the flipside I see that others disagree so I retract my suggestion here of asking for it to be done.